src/components/auth/Register.svelte

```diff
@@ -27,17 +45,35 @@
 
     <div>
       <h3 class="mb-7 text-xl font-serif">Choose a username</h3>
-      <input
-        id="registration"
-        type="text"
-        placeholder="Type here"
-        class="input input-bordered w-full block"
-        class:input-error={username.length !== 0 &&
-          (!usernameValid || !usernameAvailable)}
-        on:input={checkUsername}
-      />
+      <div class="relative">
+        <input
+          id="registration"
+          type="text"
+          placeholder="Type here"
+          class="input input-bordered w-full block"
+          class:input-error={username.length !== 0 &&
+            (!usernameValid || !usernameAvailable)}
+          on:input={checkUsername}
+        />
+        {#if checkingUsername}
+          <span
+            class="rounded-lg border-t-2 border-l-2 border-slate-600 w-4 h-4 block absolute top-4 right-4 animate-spin"
```

Comment:
So this little css beauty _approximates_ @depatchedmode's spinner, but it has some weird gradient fade thing that I don't love. Happy to convert it to an svg and use that.

Also, `animate-spin` is awesome!

Reply:
Here's what it looks like:
https://user-images.githubusercontent.com/27258/181344499-9b1b6cb3-f84c-4727-90b5-b92fcd047d5a.mov